File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.cs
public class Solution {
    public void DeleteNode(ListNode node) {
        node.val = node.next.val;
        if(node.next.next == null) node.next = null;
        else DeleteNode(node.next);
    }
}

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */

class Solution {
    /**
     * @param ListNode $node
     * @return 
     */
    function deleteNode($node) {
        $temp = $node->next;
        $node->val = $temp->val;
        $node->next = $temp->next;
    }
}